src/GoogleFit/FitOnFhir.GoogleFit/Client/Handlers/GoogleFitAuthorizationHandler.cs

```diff
@@ -75,8 +82,18 @@ private async Task<IActionResult> Authorize(RoutingRequest request)
                 return new BadRequestObjectResult($"'{Constants.PatientIdQueryParameter}' and '{Constants.SystemQueryParameter}' are required query parameters.");
             }
 
-            AuthUriResponse response = await _authService.AuthUriRequest(JsonConvert.SerializeObject(state), request.Token);
-            return new RedirectResult(response.Uri);
+            // is the token valid?
+            var isAuthenticated = await _tokenValidationService.ValidateToken(request.HttpRequest, request.Token);
```

Comment:
We might want to consider moving this check BEFORE the parameter check. I think token validation is the first thing that should be done, I think it would be odd if an unauthorized caller could get an error message for missing parameters if the token was missing or invalid.